keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/multiple/LambdaModelVertexTest.java

```diff
@@ -0,0 +1,231 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.multiple;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.algorithms.mcmc.MetropolisHastings;
+import io.improbable.keanu.algorithms.variational.optimizer.nongradient.NonGradientOptimizer;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+import io.improbable.keanu.vertices.model.LambdaModelVertex;
+import io.improbable.keanu.vertices.model.ModelVertex;
+import org.apache.commons.io.FileUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LambdaModelVertexTest {
+
+    /*
+    The model we are mimicking here lives inside a Java function, modelExecution.
+
+    It takes one input, Temperature, and produces two outputs of type double, Chance of Rain & Humidity. These outputs
+    are written to file.
+
+    It also produces one integer output and one boolean output. Suggested Factor of Suncream and 'is it sunny'. These
+    are also written to file.
+     */
+
+    private KeanuRandom random;
+    private DoubleVertex inputToModel = new ConstantDoubleVertex(50.);
```

Comment:
You never actually use the value of `50.`. If instead you initialise it to `25.` in a `@Before` method then you won't need to reinitialise `inputToModel` or `weatherModel` in your first 4 tests.

Reply:
Nice! Updated this